### PR TITLE
test: increase test coverage for logHeader and test truncate authorization header

### PR DESCRIPTION
**Pull request by szuecs**
test: increase test coverage for logHeader and test truncate authorization header

### CODE REVIEW

File: filters/diag/logheader_test.go [hunk 1]
```diff
@@ -152,3 +171,109 @@ func TestLogHeaderRequestResponse(t *testing.T) {
...
+	outputVerify.WriteString("Response for ")
+	outputVerify.WriteString(req.Method)
+	outputVerify.WriteString(" ")
+	outputVerify.WriteString(req.URL.Path)
+	outputVerify.WriteString(" ")
+	outputVerify.WriteString(req.Proto)
+	outputVerify.WriteString("\r\n")
+	outputVerify.WriteString(resp.Status)
+	outputVerify.WriteString("\r\n")
+	for k, v := range resp.Header {
```
Comment (AlexanderYastrebov) [inline filters/diag/logheader_test.go:252]: This will flake

Comment (AlexanderYastrebov) [reply filters/diag/logheader_test.go:252]: I think we have https://github.com/zalando/skipper/issues/1470 for this.
I guess we can sort headers by name before logging.

Comment (szuecs) [reply filters/diag/logheader_test.go:252]: This will only flake if there are more than one header. That's why I delete it to only have one

Comment (szuecs) [reply filters/diag/logheader_test.go:252]: You can also see bell that the following existing test has the same pattern and never appeared on our flaky test list.

File: filters/diag/logheader_test.go [hunk 2]
```diff
@@ -152,3 +171,109 @@ func TestLogHeaderRequestResponse(t *testing.T) {
...
+
+	loggerVerify = bytes.NewBuffer(nil)
+	log.SetOutput(loggerVerify)
+	log.Println(outputVerify.String())
+
+	loggerTest = bytes.NewBuffer(nil)
+	log.SetOutput(loggerTest)
+
+	lh.Response(ctx)
+	if loggerTest.Len() == 0 || !bytes.Equal(loggerTest.Bytes(), loggerVerify.Bytes()) {
```
Comment (AlexanderYastrebov) [inline filters/diag/logheader_test.go:271]: I think there is no need to check length

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**